python/ambassador/fetch/ingress.py

```diff
@@ -221,6 +235,11 @@ def _process(self, obj: KubernetesObject) -> None:
                 service_port = path_backend.get("servicePort", None)
                 path_location = path.get("path", "/")
 
+                try:
+                    service_port = int(service_port)
+                except:
+                    service_port = self._resolve_service_port_number(obj.namespace, service_name, service_port)
```

Comment:
if `_resolve_service_port_number` fails to resolve to a port number then it would return the port name if I'm reading the above method correctly right?

So later on it would default to port 80?

Reply:
that's right, in case the port can't be resolved `_resolve_service_port_number` returns the same `service_port` value. This is possible in a situation if `Service` name specified in the `Ingress` spec doesn't exist, or if the `Service` doesn't have a port with the name provided in the `Ingress`. Both cases are not liable for any ingress controller.